src/main/webapp/get-blur-areas.js

```diff
@@ -0,0 +1,264 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// suppress linter error - getBlurAreas function is used from another file
+/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "getBlurAreas" }] */
+
+'use strict';
+
+/**
+ * Function which requests server for a formUploadUrl.
+ * FormUploadUrl is the url that we will fetch to
+ * upload an image.
+ * We need this url, because we use blobstore, and this is
+ * how it works: we first upload the blob to some blobstore
+ * storage (using url from this function), blobstore does its magic,
+ * and then redirects the request to your servlet.
+ * In servlet you can access the uploaded image from that
+ * blobstore storage and not deal with unattractive real files.
+ * @return {Promise<string>} url
+ */
+function getFormUploadUrl() {
+  return new Promise(function(resolve, reject) {
+    const URL_TO_HANDLE_FORM_UPLOAD = '/get-blur-areas';
+    const FETCH_URL_TO_BLOBSTORE = '/blobstore-upload-url?forwardurl=' +
+        URL_TO_HANDLE_FORM_UPLOAD;
+
+    fetch(FETCH_URL_TO_BLOBSTORE)
+        .then(async (response) => {
+          if (!response.ok) {
+            const errorMessage = await response.text();
+            const errorText = response.status + ' server error';
+
+            alert(errorText + ' : ' + errorMessage);
+
+            reject(new Error(errorText));
+            return;
+          }
+
+          const url = await response.text();
+          resolve(url);
+        })
+        .catch((error) => {
+          alert('Unknown error while uploading image.' +
+              '\nPlease try again after some time.');
+          reject(error);
+        });
+  });
+}
+
+/**
+ * Function to send image to server and get response
+ * with areas to blur.
+ * @param {File} image
+ * @return {Promise<Array<Rect>>} blurAreas
+ */
+function getBlurAreas(image) {
+  return new Promise(async function(resolve, reject) {
+    // get new blobstore upload url.
+    const postUrl = await getFormUploadUrl().catch((error) => {
+      reject(error);
+    });
+    // if there was an error while getting form upload url.
+    if (postUrl === undefined) {
+      return;
+    }
+
+    // create form
+    const formData = new FormData();
+    formData.append('image', image);
+
+    fetch(postUrl, {
+      method: 'POST',
+      body: formData,
+    }).then(
+        (response) => {
+          (async () => {
+            if (!response.ok) {
+              const ERROR_MESSAGE = response.status + ' server error';
+
+              reject(ERROR_MESSAGE);
+
+              // show error
+              const errorMessageFromServer = await response.text();
+              alert(ERROR_MESSAGE + ' ' + errorMessageFromServer);
+
+              // throw error in order not to go to the next .then() statement
+              throw new Error(ERROR_MESSAGE);
```

Comment:
The code is similar to the one on lines 41-46. It's different, but similar :) I thought it would be nice to have similar variables 'errorMessage', 'errorText' here. It's nitpicking, but would make the code more consistent.